www/src/layouts/index.js

```diff
@@ -1,22 +1,33 @@
 import React, { useState } from "react"
+import { defaultLang, getLocaleAndBasePath } from "../utils/i18n"
 
 let PluginLibraryWrapper
-export default props => {
+export const LocaleContext = React.createContext(defaultLang)
+
+export default ({ pageContext, location, children }) => {
   const [loaded, setLoaded] = useState(false)
 
   const promise = import(`../components/layout/plugin-library-wrapper`)
-  if (props.pageContext.layout === `plugins` && !loaded) {
+  if (pageContext.layout === `plugins` && !loaded) {
     promise.then(pl => {
       PluginLibraryWrapper = pl.default
       setLoaded(true)
     })
     return null
-  } else if (props.pageContext.layout === `plugins` && loaded) {
+  } else if (pageContext.layout === `plugins` && loaded) {
     return (
-      <PluginLibraryWrapper location={props.location}>
-        {props.children}
+      <PluginLibraryWrapper location={location}>
```

Comment:
Does anyone have any context about why this is implemented this way and what this accomplishes? It seems like we're trying to lazy-load the wrapper, but can't this be done using Suspense? And why is it in this file instead of wrapping the three individual pages/templates that need this?

Reply:
This was most likely implemented long time ago (before `<Suspense>`). There is also fact that Suspense is not supported in SSR (at least last I checked) so we do need to use stuff like that anyway for now